api/src/main/java/org/openmrs/module/kenyaemr/advice/SyncMCHAppointmentsWithBahmniModule.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.kenyaemr.advice;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.Person;
+import org.openmrs.api.ConceptService;
+import org.openmrs.api.ObsService;
+import org.openmrs.api.PersonService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.appointments.model.Appointment;
+import org.openmrs.module.appointments.model.AppointmentKind;
+import org.openmrs.module.appointments.model.AppointmentProvider;
+import org.openmrs.module.appointments.model.AppointmentServiceDefinition;
+import org.openmrs.module.appointments.service.AppointmentsService;
+import org.openmrs.module.appointments.util.DateUtil;
+import org.openmrs.module.kenyaemr.metadata.MchMetadata;
+import org.springframework.aop.AfterReturningAdvice;
+import org.openmrs.module.appointments.model.AppointmentServiceType;
+import org.openmrs.module.appointments.service.AppointmentServiceDefinitionService;
+import org.openmrs.module.kenyaemr.util.EmrUtils;
+
+import java.lang.reflect.Method;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Synchronizes appointments documented in HTML forms with Bahmni appointments module
+ * Invoked after saving MCH forms
+ */
+public class SyncMCHAppointmentsWithBahmniModule implements AfterReturningAdvice {
+
+    private Log log = LogFactory.getLog(this.getClass());
+    public static final String MCH_DELIVERY_SERVICE = "b1b75503-5175-433c-a25d-763bc9650ebd";
+    public static final String MCH_POSTNATAL_VISIT_SERVICE = "dcde8ca4-32a5-4c67-9982-33346e39813f";
+    public static final String MCH_ANTENATAL_VISIT_SERVICE = "372eed95-6493-490a-8891-8374fe566aeb";
+    public static final String CWC_FOLLOWUP_SERVICE = "372eed95-6493-490a-8891-8374fe566aeb";
+    public static final String NEXT_APPOINTMENT_DATE_CONCEPT_UUID = "5096AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+    public static final String CWC_NEXT_VISIT_DATE_CONCEPT_UUID = "162549AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
```

Comment:
From this line, means CWC uses different concept for the appointment date. You need to enforce that to get the correct obs value date time. I think you have assumed all the services share the same concept for capturing next appointment date.

Reply:
All the three forms have the same concept for date except cwc